src/libs/actions/Report.js

```diff
@@ -284,7 +281,29 @@ function fetchChatReportsByIDs(chatList) {
         .then(({reports}) => {
             Log.info('[Report] successfully fetched report data', true);
             fetchedReports = reports;
-            return Promise.all(_.map(fetchedReports, updateIOUReportData));
+            return Promise.all(_.map(fetchedReports, (chatReport) => {
+                const reportActionList = chatReport.reportActionList || [];
+                const containsIOUAction = _.any(reportActionList,
+                    reportAction => reportAction.action === CONST.REPORT.ACTIONS.TYPE.IOU);
+
+                // If there aren't any IOU actions, we don't need to fetch any additional data
+                if (!containsIOUAction) {
+                    return;
+                }
+
+                // If we don't have one participant (other than the current user), this is not an IOU
```

Comment:
NAB, this comment would be better if it explained this a bit more. It's unclear how we can even reach this code.

Reply:
```
// If we don't have one participant (other than the current user), this is not an IOU
should be: 
// Group chat reports cannot and should not be associated with a specific IOU report
```

Also, I don't think we should log an alert message, because this is expected to happen. Could you simply return here?

Reply:
Or throw an error if participant.length is 0, but simple return if it is 2.

Reply:
Why would there be `0` participants? Feels like we are optimizing for a case that we may never run into.

Reply:
I've seen `0 participants` this multiple times when testing today. It's a bug which I haven't yet been able to reliably reproduce/document. It's possibly related to the issue I mentioned [here](https://github.com/Expensify/Expensify.cash/pull/2112#discussion_r615908160), but keeping the alert for now would be helpful.

Reply:
I would be happy to remove this case once the bug is resolved though.

Reply:
I updated this section to make it better.